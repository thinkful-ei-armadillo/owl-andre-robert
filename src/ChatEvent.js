import React, { Component } from 'react';
import {STORE} from './store';
//import './ParticipantList.css';
//import Participant from './Participant';

class ChatEvent extends Component {
  render() {
    let jsx = '';
    let p = STORE.participants.find(i => i.id === this.props.event.participantId);
    let time = new Date(this.props.event.time);

    switch(this.props.event.type) {
      case 'message':
        jsx = <div>
          <img src={p.avatar} width={'25px'} alt={"avatar for user"}></img>
          <div>
            <h6>{p.name}</h6> <span>{time.toString()}</span>
            <p>{this.props.event.message}</p>
          </div>
        </div>;
        break;
      default:
        //
    }

    return (
      <li className="chatEvent">{jsx}</li>
    );
  }
}

export default ChatEvent;
