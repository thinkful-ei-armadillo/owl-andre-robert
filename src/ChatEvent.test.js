import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import ChatEvent from './ChatEvent';
import { STORE } from './store';

describe('Testing Chat event generator', () => {
  // smoke test
  it('renders a message without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ChatEvent  key={'gsdfgsdfg'} event={STORE.chatEvents[0]} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  it('renders a thumbs up without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ChatEvent  key={'gsdfgsdfg'} event={STORE.chatEvents[1]} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  //snapshot test
  it('renders the UI of a message as expected', () => {
    const tree = renderer
      .create(<ChatEvent  key={'gsdfgsdfg'} event={STORE.chatEvents[0]} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders the UI of a thumbs up as expected', () => {
    const tree = renderer
      .create(<ChatEvent  key={'gsdfgsdfg'} event={STORE.chatEvents[1]} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});