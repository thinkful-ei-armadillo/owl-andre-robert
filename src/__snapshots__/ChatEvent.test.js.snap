// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Chat event generator renders the UI of a message as expected 1`] = `
<li
  className="chatEvent"
>
  <div>
    <img
      alt="avatar for user"
      src="https://robohash.org/utlaborumfugit.jpg?size=200x200&set=set1"
      width="25px"
    />
    <div>
      <h6>
        Axe Kubicka
      </h6>
       
      <span>
        Wed Jan 30 2019 07:50:46 GMT-0500 (STD)
      </span>
      <p>
        Hello world
      </p>
    </div>
  </div>
</li>
`;

exports[`Testing Chat event generator renders the UI of a thumbs up as expected 1`] = `
<li
  className="chatEvent"
>
  
</li>
`;
